Error ------------------------------------------------------------------------------------------------------ err.js:4:18

Cannot declare `f` [1] because the name is already bound. [name-already-bound]

   err.js:4:18
   4| declare function f(number): string // error
                       ^

References:
   err.js:3:10
   3| function f(x: number): number { return 42}
               ^ [1]


Error ----------------------------------------------------------------------------------------------------- scope.js:3:5

Cannot call `bar` with `0` bound to `y` because number [1] is incompatible with string [2]. [incompatible-type]

   scope.js:3:5
   3| bar(0);
          ^ [1]

References:
   scope.js:1:25
   1| declare function bar(y: T): string;
                              ^ [2]


Error ------------------------------------------------------------------------------------------------------ test.js:7:2

Cannot call `foo` because: [incompatible-type]
 - Either boolean [1] is incompatible with number [2].
 - Or boolean [1] is incompatible with string [3].

   test.js:7:2
   7| (foo(false): void); // error, boolean ~/~ undefined
       ^^^

References:
   test.js:7:6
   7| (foo(false): void); // error, boolean ~/~ undefined
           ^^^^^ [1]
   test.js:1:25
   1| declare function foo(x: number): string;
                              ^^^^^^ [2]
   test.js:2:25
   2| declare function foo(x: string): number;
                              ^^^^^^ [3]



Found 3 errors

Only showing the most relevant union/intersection branches.
To see all branches, re-run Flow with --show-all-branches
